sycl/doc/extensions/experimental/sycl_ext_oneapi_non_uniform_groups.asciidoc

```diff
@@ -423,225 +424,102 @@ if (will_branch)
+  // get the set of work-items that called this function simultaneously
+  auto active_group = sycl::ext::oneapi::experimental::this_kernel::get_opportunistic_group();
 
-[source, c++]
-----
-auto sg = it.get_sub_group();
+  // increment the atomic once on behalf of all active work-items
+  int count = active_group.get_local_linear_range();
+  int old_value;
+  if (active_group.leader()) {
+    old_value = ptr.fetch_add(count);
+  }
 
-// reduce over contiguous groups of 8 elements
-auto partition = sycl::ext::oneapi::experimental::get_fixed_size_group<8>(sg);
-auto result = sycl::reduce_over_group(partition, buf[it.get_local_linear_id()], sycl::plus<>());
+  // return the value the individual work-item might have received if it had worked alone
+  auto base = sycl::group_broadcast(active_group, old_value);
+  auto idx = active_group.get_local_linear_id();
+  return base + idx;
 
-// write result out once per group
-if (partition.leader()){
-  buf[partition.get_group_id()] = result;
 }
 ----
 
-A `fixed_size_group` can be used to provide an interface accepting a specific
-number of work-items:
 
-[source, c++]
-----
-void func_that_needs_4_work_items(sycl::ext::oneapi::experimental::fixed_size_group<4> group);
-----
+=== Chunk
 
-
-=== Tangle-Groups
-
-A tangle-group is a non-contiguous subset of a group representing work-items
-executing in a tangle. A tangle-group can therefore be used to capture all
-work-items currently executing the same control flow.
+A `chunk` represents a contiguous collection of work-items created by
+subdividing a group into equally sized parts, such that each work-item is a
+member of exactly one partition.
+The size of a chunk is a static (compile-time) property.
 
 
 ==== Creation
 
-New tangle-groups are created by partitioning an existing group, using the
-`get_tangle_group` free-function.
-
-NOTE: Creating a tangle-group may implicitly synchronize members of the
-`tangle_group` on some devices, since it may be necessary to wait for
-work-items to reconverge. For consistency, this synchronization is required by
-all implementations.
+A new `chunk` can only be created by partitioning an existing group with the
+`chunked_partition` function.
 
 [source, c++]
 ----
 namespace ext::oneapi::experimental {
 
-template <typename Group>
-tangle_group get_tangle_group(Group group);
+template <size_t ChunkSize, typename ParentGroup>
+chunk<ChunkSize, Group> chunked_partition(ParentGroup parent);
 
 } // namespace ext::oneapi::experimental
 ----
 
-_Constraints_: Available only if `sycl::is_group_v<std::decay_t<Group>> &&
-std::is_same_v<Group, sycl::sub_group>` is true.
+_Constraints_: Available only if `sycl::is_group_v<std::decay_t<ParentGroup>>`
+is true and `ParentGroup` is a `sycl::sub_group` or `sycl::chunk`.
+`ChunkSize` must be positive and a power of 2.
+
+_Preconditions_: `ChunkSize` must be less than or equal to the result of
+`parent.get_max_local_range()`.
```

Comment:
`chunk` does not define `get_max_local_range()`, but it can be passed as `parent`.

`sub_group::max_local_range()` returns `range`, but `ChunkSize` is `size_t`, we can't compare them, should we use `parent.get_max_local_range().size()` instead?

Reply:
I think this is just a bug, and it probably should have been `parent.get_local_linear_range`.  The second sentence about divisibility covers this, so I've just removed this sentence in 6679e8103b119d69b86f67a220a4b238f4c96ca9.